spectator-web-spring/src/main/java/com/netflix/spectator/controllers/MetricsController.java

```diff
@@ -0,0 +1,229 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spectator.controllers;
+
+import com.netflix.spectator.api.Counter;
+import com.netflix.spectator.api.DistributionSummary;
+import com.netflix.spectator.api.Gauge;
+import com.netflix.spectator.api.Id;
+import com.netflix.spectator.api.LongTaskTimer;
+import com.netflix.spectator.api.Measurement;
+import com.netflix.spectator.api.Meter;
+import com.netflix.spectator.api.Registry;
+import com.netflix.spectator.api.Timer;
+
+import com.netflix.spectator.controllers.filter.ChainedMeasurementFilter;
+import com.netflix.spectator.controllers.filter.MeasurementFilter;
+import com.netflix.spectator.controllers.filter.PrototypeMeasurementFilter;
+import com.netflix.spectator.controllers.filter.TagMeasurementFilter;
+import com.netflix.spectator.controllers.model.ApplicationRegistry;
+import com.netflix.spectator.controllers.model.MetricValues;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnExpression;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+import java.io.IOException;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.HashMap;
+
+
+/**
+ * Provides an HTTP endpoint for polling spectator metrics.
+ */
+@RequestMapping("/spectator/metrics")
+@RestController
+@ConditionalOnExpression("${spectator.endpoint.enabled:false}")
+public class MetricsController {
+  @Autowired
+  private Registry registry;
+
+  @Value("${spectator.applicationName:}")
+  private String applicationName;
+
+  @Value("${spectator.applicationVersion:}")
+  private String applicationVersion;
+
+  @Value("${spectator.endpoint.tagFilter.meterNameRegex:.*}")
+  private String meterNameRegex;
+
+  @Value("${spectator.endpoint.tagFilter.tagNameRegex:}")
+  private String tagNameRegex;
+
+  @Value("${spectator.endpoint.tagFilter.tagValueRegex:}")
+  private String tagValueRegex;
+
+  @Value("${spectator.endpoint.prototypeFilter.path:}")
+  private String prototypeFilterPath;
+
+  private MeasurementFilter defaultMeasurementFilter = null;
+
+  /**
+   * The default measurement filter is configured through properties.
```

Comment:
Agree this is awkward. Is there a simple way to do this that wouldnt be ambiguous?
I dont think I should combine them (if both are specified then chain them together). The tag filters are there by default and I'd configure them to say keep everything that has labels (.+) which filters out the spring boot added metrics that we dont want in practice but seems wrong to hard code in. I guess I instead have a flag for that particular case (but question of when or how to override it remains and is more foggy)

BTW, I dont filter out the non-labeled things here, but could. Should I? I didnt want to impose that policy here unless it is a general recommendation to spectator consumers.

The profile filter is there to make it easy to build a profile for just the things you care about, especially if you are pushing metrics into another system and dont want to clog it up with stuff you dont care about. I figured if you used that mechanism, you dont need the crude tag mechanism.

On the query side, the tags seem more useful since they are just one-off usage to spot check something narrow.